yt/frontends/enzo_e/io.py

```diff
@@ -19,11 +19,16 @@ def __init__(self, *args, **kwargs):
             slice(self.ds.ghost_zones, -self.ds.ghost_zones),
         )
 
-        # Determine if particle masses are actually densities by the
-        # existence of the "mass_is_mass" particles parameter.
-        mass_flag = nested_dict_get(
-            self.ds.parameters, ("Particle", "mass_is_mass"), default=None
-        )
+        # Determine if particle masses are actually masses or densities.
+        if self.ds.parameters['version_string'] is not None:
+            # they're masses for enzo-e versions that record a version string
+            mass_flag = True
+        else:
+            # in earlier versions: query the existence of the "mass_is_mass"
+            # particle parameter
+            mass_flag = nested_dict_get(
+                self.ds.parameters, ("Particle", "mass_is_mass"), default=None
+            )
         self._particle_mass_is_mass = mass_flag is not None
```

Comment:
I know you haven't touched this line here but would it make more sense as follows ?
```suggestion
        self._particle_mass_is_mass = bool(mass_flag)
```

Reply:
The way it currently reads just checks for the existence of the parameter. For better or worse, the way that parameter historically got set was unsound and could set the parameter to a random value. This includes a value of zero. So I think we need to keep the current logic.

Reply:
Alright that makes sense to me. If you have a minute to add this information as a comment I think that'd be very valuable :)